demos/text_detection_demo/src/text_detection.cpp

```diff
@@ -98,7 +98,36 @@ std::vector<cv::RotatedRect> maskToBoxes(const cv::Mat &mask, float min_area, fl
     }
 
     return bboxes;
-  }
+}
+
+std::vector<cv::RotatedRect> coordToBoxes(const float* coords,
+                                          const size_t &size_data,
+                                          float min_area, float min_height,
+                                          const cv::Size &input_shape,
+                                          const cv::Size &image_size) {
+    std::vector<cv::RotatedRect> bboxes;
+    int max_bbox_idx = size_data / 5;
+    float x_scale = image_size.width / float(input_shape.width);
+    float y_scale = image_size.height / float(input_shape.height);
+
+    for (int i = 1; i <= max_bbox_idx; i++) {
+        const float *prediction = &coords[(i - 1) * 5];
```

Comment:
```suggestion
    for (int i = 0; i < max_bbox_idx; i++) {
        const float *prediction = &coords[i * 5];
```

Seems like that would be a more conventional loop.